ext/compressors/iaa/iaacodec/include/IAACompressionCodecDeflate.h

```diff
@@ -0,0 +1,218 @@
+            }
+            *job_id = jobPoolSize - index;
+            return jobPool[index];
+        }
+        else
+        {
+            return nullptr;
+        }
+    }
+    inline qpl_job * releaseJob(uint32_t job_id)
+    {
+        if(jobPoolEnabled)
+        {
+            uint32_t index = jobPoolSize - job_id;
+            ReleaseJobObjectGuard _(index);
+            return jobPool[index];
+        }
+        else
+        {
+            return nullptr;
+        }
+    }
+    inline qpl_job * getJobPtr(uint32_t job_id)
+    {
+        if(jobPoolEnabled)
+        {
+            uint32_t index = jobPoolSize - job_id;
+            return jobPool[index];
+        }
+        else
+        {
+            return nullptr;
+        }
+    }
+
+private:
+
+    inline int32_t get_job_size_helper()
+    {
+        static uint32_t size = 0;
+        if (size == 0)
+        {
+            const auto status = qpl_get_job_size(PATH, &size);
+            if (status != QPL_STS_OK)
+            {
+                return -1;
+            }
+        }
+        return size;
+    }
+
+    inline int32_t init_job_helper(qpl_job * qpl_job_ptr)
+    {
+        if (qpl_job_ptr == nullptr)
+        {
+            return -1;
+        }
+        auto status = qpl_init_job(PATH, qpl_job_ptr);
+        if (status != QPL_STS_OK)
+        {
+            return -1;
+        }
+        return 0;
+    }
+
+    inline int32_t initJobPool()
+    {
+        static bool initialized = false;
+
+        if (initialized == false)
+        {
+            const int32_t size = get_job_size_helper();
+            if (size < 0)
+                return -1;
+            for (int i = 0; i < jobPoolSize; ++i)
+            {
+                jobPool[i] = nullptr;
+                qpl_job * qpl_job_ptr = reinterpret_cast<qpl_job *>(new uint8_t[size]);
+                if (init_job_helper(qpl_job_ptr) < 0)
+                    return -1;
+                jobPool[i] = qpl_job_ptr;
+                jobLock[i].store(false);
+            }
+            initialized = true;
+        }
+        return 0;
+    }
+
+    inline bool tryLockJob(size_t index)
+    {
+        bool expected = false;
+        return jobLock[index].compare_exchange_strong(expected, true);
+    }
+
+    inline void destroyJobPool()
+    {
+        const uint32_t size = get_job_size_helper();
+        for (uint32_t i = 0; i < jobPoolSize && size > 0; ++i)
+        {
+            while (tryLockJob(i) == false)
+            {
+            }
+            if (jobPool[i])
+            {
+                qpl_fini_job(jobPool[i]);
+                delete[] jobPool[i];
+            }
+            jobPool[i] = nullptr;
+            jobLock[i].store(false);
+        }
+    }
+
+    struct ReleaseJobObjectGuard
+    {
+        uint32_t index;
+        ReleaseJobObjectGuard() = delete;
+
+    public:
+        inline ReleaseJobObjectGuard(const uint32_t i) : index(i)
+        {
+            //nothing to do
```

Comment:
Please start all comments with a capital letter and end with a full stop.

Reply:
OK, We will modify it. Thanks!